lib/Interpreter/CppInterOp.cpp

```diff
@@ -2866,9 +2877,10 @@ namespace Cpp {
     if (auto* FunctionTemplate = dyn_cast<FunctionTemplateDecl>(TemplateD)) {
       FunctionDecl* Specialization = nullptr;
       clang::sema::TemplateDeductionInfo Info(fakeLoc);
-      if (Sema::TemplateDeductionResult Result = S.DeduceTemplateArguments(
-              FunctionTemplate, &TLI, Specialization, Info,
-              /*IsAddressOfFunction*/ true)) {
+      Template_Deduction_Result Result = S.DeduceTemplateArguments(
+          FunctionTemplate, &TLI, Specialization, Info,
+          /*IsAddressOfFunction*/ true);
+      if (static_cast<int>(Result)) {
```

Comment:
```suggestion
      if (Result != TemplateDeductionResult::Success) {
```

I think the static_cast here is suboptimal.